### PR TITLE
Reformat migrations and implement cleaner dry run behavior

### PR DESCRIPTION
**Pull request by eggplantzzz**
Remove `skipDryRun` from the props in the config so that it can be set on the command line as a flag.

### CODE REVIEW

**Comment by eggplantzzz** [comment]
I removed the getter/setter bit of code from the `props` object. Please let me know if you think this might cause problems somewhere else in the codebase.

**Comment by eggplantzzz** [comment]
You should still be able to set that option in the config. In the `detect` method it will still merge in `static_config` which is the values from the Truffle config.

One thing I've noted is the inconsistency with how the config is being initialized with each command. In other words, sometimes the config is initialized with `default`, some with `detect`, and others with `default().with({ ... })`. As long as the command initializes the config with something like `detect`, it should import all of the Truffle config settings. In this particular case, however, it is most likely fine since `--skip-dry-run` only applies to migrations.

**Comment by CruzMolina** [comment]
Hmm, well after demoing the branch locally, it appears I'm unable to set the option in the config.

Can you add a test or verify on your end?

**Comment by gnidan** [comment]
(removing myself from request for review cause y'all have it covered :)

**Comment by eggplantzzz** [comment]
@CruzMolina So currently it will skip the dry run if you configure the variable in camel case but it will not accept it in kebab case like you would specify from the command line. It think yargs does some magic to make both versions of the property internally available to Truffle. I'll make some changes in order to handle both version.

**Comment by eggplantzzz** [comment]
@CruzMolina I think the behavior you mentioned before is all cleared up. I also reworked a bunch of the functions in truffle-migrate to not take the callbacks that they were previously passed. I pulled it all "high up" in the command flow. If I changed any logic it was unintentional. Could you look at it with an eye out for that?

File: packages/truffle-core/lib/commands/migrate.js [hunk 1]
```diff
@@ -110,6 +112,62 @@ var command = {
       61717561 // Aquachain
     ];
 
+    let dryRunOnly;
```
[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/truffle-core/lib/commands/migrate.js]: Not sure if the above actually helps resolve the issue.

[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/truffle-core/lib/commands/migrate.js]: Yep, perfect. That was the culprit. Good catch!

[the remark this answers is not in the document]
Comment (CruzMolina) [reply packages/truffle-core/lib/commands/migrate.js]: Hmm, still the same issue locally.

[the remark this answers is not in the document]
Comment (eggplantzzz) [reply packages/truffle-core/lib/commands/migrate.js]: @CruzMolina Ok, I think I found a problem with the way a callback was being called. It looks good on my local, can you verify one more time for me please?

**Comment by CruzMolina** [comment]
Hooray! 👏

**Pull request merged**